### PR TITLE
Fix t/ui/18-test-details.t

### PR DESCRIPTION
**Pull request by Martchus**
The test seems stable so I removed the `DO NOT MERGE: Test stability of t/ui/18-tests-details.t` commit. See the particular commit messages of the remaining commits for details.

### CODE REVIEW

File: t/lib/OpenQA/Test/Utils.pm [hunk 1]
```diff
@@ -632,7 +633,8 @@ sub test_cmd {
     return $ret;
 }
 
-sub wait_for_or_bail_out(&*;*) {
+no feature 'signatures';    # `&*;*` syntax is not compatible with signatures
```
Comment (okurz) [inline t/lib/OpenQA/Test/Utils.pm:636]: I would consider it cleaner to put that into a block, i.e.

```perl
{
no feature ...
...
}}}
```

File: t/ui/18-tests-details.t [hunk 2]
```diff
@@ -446,9 +446,7 @@ subtest 'misc details: title, favicon, go back, go to source view, go to log vie
         'on src page for installer_timezone test'
     );
     is($driver->find_element('.cm-comment')->get_text(), '#!/usr/bin/env perl', 'we have a perl comment');
-    $driver->go_back();    # to 99946
-    $driver->find_element_by_link_text('Logs & Assets')->click;
-    wait_for_ajax msg => 'logs and assets tab';
+    $driver->get('/tests/99946/downloads_ajax');
```
Comment (okurz) [inline t/ui/18-tests-details.t:449]: As you stated this to be a workaround I suggest to put the explanation from the git commit message as comment in here

**Review by okurz — APPROVED** [review]

File: t/ui/12-needle-edit.t [hunk 3]
```diff
@@ -96,22 +96,15 @@ shared_hash {};
 my $git_mock = Test::MockModule->new('OpenQA::Git');
 $git_mock->redefine(commit => sub ($self, $args) { shared_hash $args; return undef });
 
-# prepare clean needles directory
-my $dir = path('t/data/openqa/share/tests/opensuse/needles')->remove_tree->make_path;
-
 plan skip_all => $OpenQA::SeleniumTest::drivermissing unless my $driver = call_driver({with_gru => 1});
 
+# prepare clean needles directory, create default 'inst-timezone' needle
+my $dir              = prepare_clean_needles_dir;
```
Comment (asdil12) [inline t/ui/12-needle-edit.t:102]: Maybe use the name `$needle_dir` here as well analog to `t/ui/18-tests-details.t`

**Review by asdil12 — APPROVED** [review]
LGTM overall

**Pull request merged**